### ISSUE TITLE
[coc.nvim] Command: ccls.restart not found

### ISSUE THREAD

**Issue by atomashpolskiy** [neoclide/coc.nvim#1094]
Hi guys! I'm having a weird issue with `coc-ccls` commands being present in the CocList->Commands list, but not being able to invoke them.

## versions

vim version: NVIM v0.3.8
node version: v12.8.1
coc.nvim version: 0.0.74-bc017b02e6
term: alacritty
platform: darwin

## Messages
[coc.nvim] Command: ccls.callHierarchy not found
[coc.nvim] Command: ccls.callHierarchy not found
## Output channel: languageserver.ccls

20:23:42                initialize.cc:271 I initialize in directory /Users/sadpotato/git-workspace/dropbox-json11 with uri file:///Users/sadpotato/git-workspace/dropbox-json11
20:23:42                initialize.cc:294 I initializationOptions: {"compilationDatabaseCommand":"","compilationDatabaseDirectory":"","cache":{"directory":"/tmp/ccls","format":"binary","hierarchicalPath":false,"retainInMemory":2},"capabilities":{"documentOnTypeFormattingProvider":{"firstTriggerCharacter":"}","moreTriggerCharacter":[]},"foldingRangeProvider":true,"workspace":{"workspaceFolders":{"supported":true,"changeNotifications":true}}},"clang":{"excludeArgs":[],"extraArgs":[],"pathMappings":[],"resourceDir":""},"client":{"diagnosticsRelatedInformation":true,"hierarchicalDocumentSymbolSupport":true,"linkSupport":true,"snippetSupport":true},"codeLens":{"localVariables":true},"completion":{"caseSensitivity":2,"detailedLabel":true,"dropOldRequests":true,"duplicateOptional":true,"filterAndSort":true,"include":{"blacklist":[],"maxPathSize":30,"suffixWhitelist":[".h",".hpp",".hh",".inc"],"whitelist":[]},"maxNum":100},"diagnostics":{"blacklist":[],"onChange":1000,"onOpen":0,"onSave":0,"spellChecking":true,"whitelist":[]},"highlight":{"largeFileSize":2097152,"lsRanges":true,"blacklist":[],"whitelist":[]},"index":{"blacklist":[],"comments":2,"initialNoLinkage":false,"initialBlacklist":[],"initialWhitelist":[],"maxInitializerLines":5,"multiVersion":0,"multiVersionBlacklist":[],"multiVersionWhitelist":[],"name":{"suppressUnwrittenScope":false},"onChange":false,"parametersInDeclarations":true,"threads":0,"trackDependency":2,"whitelist":[]},"request":{"timeout":5000},"session":{"maxNum":10},"workspaceSymbol":{"caseSensitivity":1,"maxNum":1000,"sort":true},"xref":{"maxNum":2000}}
20:23:42                initialize.cc:323 I use -resource-dir=/usr/local/Cellar/llvm/8.0.0_1/lib/clang/8.0.0
20:23:42                initialize.cc:356 I workspace folder: /Users/sadpotato/git-workspace/dropbox-json11/
20:23:42                   project.cc:397 I loaded /Users/sadpotato/git-workspace/dropbox-json11/compile_commands.json
20:23:42                initialize.cc:381 I start 8 indexers
20:23:42                initialize.cc:389 I dispatch initial index requests
20:23:42                  pipeline.cc:477 I loaded project. Refresh semantic highlight for all working file.
20:23:42              sema_manager.cc:734 I create session for /Users/sadpotato/git-workspace/dropbox-json11/json11.cpp
20:23:42                  pipeline.cc:343 I parse /Users/sadpotato/git-workspace/dropbox-json11/json11.cpp
20:23:42                  pipeline.cc:293 I load cache for /Users/sadpotato/git-workspace/dropbox-json11/json11.cpp
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_types/_uint8_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_ct_rune_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/i386/types.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/signal.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/stdlib.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/stdint.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_ptrdiff_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_errno_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_types/_uint64_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/cstdint (delta: 1)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tuple (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/machine/types.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_types/_uint32_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_wint_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_endian.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/cstring (delta: 1)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__nullptr (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__debug (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_wchar_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/wait.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_size_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/Availability.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_rsize_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_intptr_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_u_int64_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_u_int32_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_int8_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_int64_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_int16_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_types/_uint16_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_mode_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_offsetof.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/new (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__string (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/machine/_types.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_int32_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_types.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/mach/i386/_structs.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/cstddef (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/i386/_types.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__config (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/initializer_list (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h (delta: 1)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_ssize_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stddef.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_posix_availability.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_types/_u_int8_t.h (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Users/sadpotato/git-workspace/dropbox-json11/json11.hpp (delta: 0)
20:23:43                  pipeline.cc:381 I store index for /Users/sadpotato/git-workspace/dropbox-json11/json11.cpp (delta: 0)

## Describe the bug

`:CocList -> commands` menu shows ccls commands such as `restart`, `callHierarchy`, etc., but invoking these commands produces message `Command: ccls.<command name> not found`

## Screenshots

https://imgur.com/a/c9rQo0u

## Config

```
$ cat ~/.config/nvim/coc-settings.json
{
    "languageserver": {
        "ccls": {
            "command": "ccls",
            "filetypes": ["c", "cpp", "objc", "objcpp"],
            "rootPatterns": [".ccls", "compile_commands.json", ".vim/", ".git/", ".hg/"],
            "initializationOptions": {
                "cache": {
                    "directory": "/tmp/ccls"
                },
                "highlight": { "lsRanges" : true }
            }
        }
    }
}
```

**Comment by chemzqm** [comment]
There's no way for configured language server to register CocCommand.
You can restart coc.nvim with command `:CocRestart`

**Comment by atomashpolskiy** [comment]
Thank you for your response! May I ask two questions though?

1) Why are these commands present in the commands menu? Should I just
ignore them?

2) Is there any way to actually run them? Like :call CocAction(...) or
something like that?

**Comment by chemzqm** [comment]
1. Your coc extension
2. Yes, read the doc

**Comment by atomashpolskiy** [comment]
I've spent some time reading the doc.txt, and this seems to be the intended way of running LS commands, right?

```
"runCommand" [{name}] [...{args}] 		*coc-action-runCommand*

	Run a global command provided by the language server. If {name} is not 
	provided, a prompt with a list of commands is shown to be selected.

	{args} are passed as arguments of command.

	You can bind your custom command like so: >

	command! -nargs=0 OrganizeImport
		\ :call CocActionAsync('runCommand', 'tsserver.organizeImports')
```

Now, when I use `:call CocAction('runCommand')`, it shows the same list as when using `:CocList`. I select one of ccls commands and press Enter, and the same error appears.

Same error is shown, when I just type the non-interactive version `:call CocAction('runCommand', 'ccls.callHierarchy')`.

Perhaps I'm missing something obvious... I would be really grateful, if you pointed me to the correct way of running a language server command!

**Comment by chemzqm** [comment]
ccls.callHierarchy is a CocCommand, it's a request method.

**Comment by atomashpolskiy** [comment]
Figured this out, thank you! Using `coc-ccls` was one of the reasons for my confusion, because it seems to register new commands, which can't be run. Turns out that coc.nvim itself provides complete integration with ccls, so I uninstalled coc-ccls (doing so removed the ccls.* commands from the CocList->commands menu).

Now I'm using the following command, and everything works quite nicely:

```
:call CocLocations('ccls', '$ccls/call', {'callee': v:false, 'hierarchy': v:true}, 'edit')
```

Definitely will be learning more from https://github.com/MaskRay/ccls/wiki/FAQ and coc.nvim's doc.txt. Nvim+Coc is pure awesomeness, thank you very much for building this plugin! :)
